### ISSUE TITLE
"Not Now" does nothing

### ISSUE THREAD

**Issue by david32** [gioxx/MarvellousSuspender#86]
* Extension version: v7.1.6.2
 * Browser name & version: Chrome Version 89.0.4389.114 
 * Operating system & version: Windows 10 20H2

When deciding not to suspend a certain tab, and clicking on the "Not Now" button in the extension window, nothing happens.
![image](https://user-images.githubusercontent.com/2933880/113557388-9c3ac700-9606-11eb-8c91-608a5a9b8922.png)

I could help provide some logs or something if you tell me how it should be done.

Also, thanks for taking control of this extension!

**Comment by MattSuda** [comment]
I'm having the same problem. Same extension and Chrome version on Mac

**Comment by gioxx** [comment]
Bug verified, sorry for the inconvenience. We will fix this in a next version.
edit: If you click on "Not Now" immediately, this does nothing. If you change tab and return to the previous, you can click "Not Now" and this works well, very very strange, this behavior needs to be better investigated.

**Comment by baz357** [comment]
> edit: If you click on "Not Now" immediately, this does nothing. If you change tab and return to the previous, you can click "Not Now" and this works well, very very strange

This workaround only works if there's another tab in the current window. Very strange indeed! :)

**Comment by david32** [comment]
I was going to write that this workaround doesn't work for me, but it seems that it does. The icon changes to gray but the text remains the same.

That is very good to know, thank you!

**Comment by mkz7** [comment]
> If you click on "Not Now" immediately, this does nothing. If you change tab and return to the previous, you can click "Not Now" and this works well, very very strange, this behavior needs to be better investigated.

That's the behavior I've been seeing: 
'Change tab and then return to' tab that previously didn't respond to "Not Now." 
Then it works. 
This started happening as of the 7.1.6.2 update.

TMS 7.1.6.2
Brave browser v. 1.21.73
OS Mojave 10.14.3

**Comment by PepeParada** [comment]
Tengo el mismo problema. "Not now" no funciona. Hago click, pero no sucede nada.
--
-I have the same problem. "Not now" doesn't work. I click, but nothing happens.

Chrome Versión 90.0.4430.93 (Build oficial) (64 bits)
Windows 7 
TMS Versión 7.1.6.2

**Comment by S1rK** [comment]
I also have the same issue.
When is it gonna be fixed (the issue has been reported almost 2 months ago)?


TMS Version 7.1.6.2
Chrome Version 90.0.4430.212 (Official Build) (64-bit)
Windows 10 Version 10.0.19041 Build 19041

**Comment by gioxx** [comment]
Ciao @S1rK , I can understand the frustration but, since I was infected by Covid-19 and @sherkot was "temporarily out of the game" because he has meanwhile married, We haven't had a chance to rework the code to fix the various problems you have reported.

This will change shortly, we already agreed to work on the reported problems, to release a new version of TMS that will correct them and introduce some of the requested improvements.

Thanks for your patience, **I remind everyone that the code is completely open and is welcome the help of other developers who want to try to solve the issues reported**.

**Comment by S1rK** [comment]
> Ciao @S1rK , I can understand the frustration but, since I was infected by Covid-19 and @sherkot was "temporarily out of the game" because he has meanwhile married, We haven't had a chance to rework the code to fix the various problems you have reported.
> 
> This will change shortly, we already agreed to work on the reported problems, to release a new version of TMS that will correct them and introduce some of the requested improvements.
> 
> Thanks for your patience, **I remind everyone that the code is completely open and is welcome the help of other developers who want to try to solve the issues reported**.


First of all, @gioxx, take care of your health and I hope you'll get better soon!
Second, congrats @sherkot for your marriage!
I'm sure we all can wait longer, there is no pressure on you guys.
I don't have any knowledge in extensions programming, so unfortunately I can't help.

**Comment by mkz7** [comment]
Hope you are well now, @gioxx. (And @sherkot still happily married.)

This seems like a good time to ask what "Not Now" actually means. That is, when does "Not Now" revert to the user's chosen default setting? 

An actual definition of "Not Now" would be useful to have to better determine whether, in specific situations, it's the better option to choose, rather than the more permanent one ("Never suspend this URL").

Also suggest that the "_Suspend all other tabs_" and "_Unsuspend all tabs_" options include the words "_in this window_" since otherwise "all" would tend to mean "in all open windows."

Thanks.

**Comment by mkz7** [comment]
@gioxx @sherkot
No sign that this has been 'fixed' (https://github.com/gioxx/MarvellousSuspender/issues/86). 
When I click on "Not Now" it's the same as before. Not recognized.

Also, before 'closing' this, a reply to the questions in my post would be appreciated. 
I haven't seen "Not Now" explained anywhere and would appreciate knowing how it works in practice.

Thanks.

**Comment by baz357** [comment]
> Also suggest that the "_Suspend all other tabs_" and "_Unsuspend all tabs_" options include the words "_in this window_" since otherwise "all" would tend to mean "in all open windows."

Yeah, I've avoided using that option for exactly that reason: not _knowing_ what it does (with my tons of windows and tabs) made it seem like a potentially dangerous option. ;)

**Comment by d4h0** [comment]
@mkz7: The fix first needs to be published in a release on the extension store. Did you install the extension from the source code in this repo?

"Not Now" means, "Don't suspend this tab now", and that is obvious if you think about what the extension does.

Edit: deleted something regarding something that I might have misinterpreted...

**Comment by mkz7** [comment]
@sherkot You are reading something into my post which wasn't in it. There was no rudeness present or intended.  

It appears I wasn't clear about my question re: the meaning of "Not Now."

What I meant to ask was what the '**technical**' meaning of that condition is -- that is, yes it's clear that it means that at the current time ("Now") the tab won't suspend -- but at what point, or under what conditions, does a "Not Now" tab revert to the user's default setting for when tabs do suspend (e.g., after one hour of inactivity, etc.)? 

To put it another way, what exactly makes a "Not Now" tab suspend without any user intervention.
It makes sense that "Not Now" is temporary, since otherwise the setting would be no different from "Never suspend"'
But what triggers the tab's reverting to a 'suspend' state isn't stated anywhere as far as I'm aware.

Thanks.

**Comment by mkz7** [comment]
Hi @d4h0, you are reading something into my post which wasn't in it. There was no rudeness present or intended.  

It appears I wasn't clear about my question re: the meaning of "Not Now."

What I meant to ask was what the '**technical**' meaning of that condition is -- that is, yes it's clear that it means that at the current time ("Now") the tab won't suspend -- but at what point, or under what conditions, does a "Not Now" tab revert to the user's default setting for when tabs do suspend (e.g., after one hour of inactivity, etc.)? 

To put it another way, what exactly makes a "Not Now" tab suspend without any user intervention. It makes sense that "Not Now" is temporary, since otherwise the setting would be no different from "Never suspend". But what triggers the tab's reverting to a 'suspended' state isn't stated anywhere as far as I'm aware.

Thanks.

    On Friday, May 28, 2021, 02:35:42 PM EDT, d4h0 ***@***.***> wrote:  
 
 


@mkz7: The fix first needs to be published in a release on the extension store. Did you install the extension from the source code in this repo?

"Not Now" means, "Don't suspend this tab now", and that is pretty obvious if you think about what the extension does.

Besides that: Your message seems pretty rude. You should think about how you talk to people who work for free on projects you use. This behavior is discouraging to the maintainers.

**Comment by d4h0** [comment]
> Hi @d4h0, you are reading something into my post which wasn't in it. There was no rudeness present or intended.

@mkz7: Alright. After publishing my comment and then rereading what you wrote, I came to the same conclusion (see my edit above).

> at what point, or under what conditions, does a "Not Now" tab revert to the user's default setting for when tabs do suspend (e.g., after one hour of inactivity, etc.)? 

Okay, that wasn't clear to me. But I wondered the same in the past. I never noticed that "Not now" was reverted automatically – but I'm pretty sure that I read somewhere that it would. So a clarification would be great.

> To put it another way, what exactly makes a "Not Now" tab suspend without any user intervention.

I'm pretty sure, restarting the browser does clear the "Not Now" mark of a tab. But, as I wrote, some documentation would be great.

**Comment by mkz7** [comment]
@gioxx @sherkot Would appreciate your clarifying the 'Closed' status. 

If the issue has been fixed will the update with the fix be pushed to users who have TMS installed via the Chrome Web Store?

So far I see no change in the behavior of "Not Now" (i.e., it doesn't work).

Thanks.

**Comment by gioxx** [comment]
Ciao @mkz7, I can confirm we have solved the problem in the GitHub version of TMS, you have to wait "public" TMS 7.1.6.3 if you use the Chrome Store version (and we want to solve other problems and introduce Tab Group's save / export before push the new version on Google, we hope as soon as possible).

In the meantime, I can reply to the doubt about "Not Now" function. "Not Now" use a temporary whitelist who permit to mantain open and not suspended tabs. If you exit from the browser, tabs previously set to "Not Now" will reset and get suspended as usual. The same behaviour you can obtain if you close the tab and open again (without using CTRL+SHIFT+T). If you want to save this condition, you have to use "Never suspend this url" (or domain).

If you want some other technical details, I think @sherkot can reply better than me.

**Comment by ossilator** [comment]
"do not suspend this tab" would be clear enough in both spatial and temporal scope. the problem is that this wouldn't really fit with the preceding part of the menu item, which is actually a status message. i guess one could address this by separating the two, thus turning the status into a passive title, and having a proper context-sensitive additional menu item right below it instead.

**Comment by mkz7** [comment]
Thanks, @gioxx (cc: @sherkot).

_> you have to wait "public" TMS 7.1.6.3 if you use the Chrome Store version (and we want to solve other problems and introduce Tab Group's save / export before push the new version on Google, we hope as soon as possible)._

Is there a way to install the update now from Github? I'm not particularly 'technical' (as most users here seem to be) but I can follow instructions if the procedure is a simple one.
 
_> "Not Now" use a temporary whitelist who permit to mantain open and not suspended tabs. If you exit from the browser, tabs previously set to "Not Now" will reset and get suspended as usual._ 

Okay, so the only difference is that "Never suspend" persists following a browser exit, whereas "Not Now" doesn't?

**Comment by gioxx** [comment]
> 
> 
> Is there a way to install the update now from Github? I'm not particularly 'technical' (as most users here seem to be) but I can follow instructions if the procedure is a simple one.

Of course!
If you want to use the GitHub version, you need to uninstall the "public" (Chrome Store) version before. **Please, pay attention**: unsuspend all tabs or export a backup before the uninstallation, otherwise you will lose all the tabs.

Now you can download the current TMS (updated on GitHub) via https://github.com/gioxx/MarvellousSuspender/archive/refs/heads/master.zip. Unzip the archive and load an unpacked extension in Chrome:

- Open a new tab in Chrome and go to chrome://extensions
- Enable Developer Mode (in the upper right corner)
- Load unpacked extension (button in the upper left corner)
- Load the "src" folder you have previously unzipped (only the "src", not the entire TMS GitHub folder)

Now you have TMS 7.1.6.2 installed from GitHub, the modified version (actually a work in progress), you can suspend all tabs again or (if you have lose some tabs), you can import the saved session

> 
> 
> Okay, so the only difference is that "Never suspend" persists following a browser exit, whereas "Not Now" doesn't?

Exactly.

**Comment by gioxx** [comment]
> 
> 
> "do not suspend this tab" would be clear enough in both spatial and temporal scope. the problem is that this wouldn't really fit with the preceding part of the menu item, which is actually a status message. i guess one could address this by separating the two, thus turning the status into a passive title, and having a proper context-sensitive additional menu item right below it instead.

Is a good suggestion that we will evaluate when improving the extension, thanks.

**Comment by mkz7** [comment]
_> Of course!_

Thanks a lot for those instructions, @gioxx. 

Since this isn't a major issue for me (functionality of "Not Now" and "Never" isn't radically different) I may just wait for the Chrome Web Store version.

Do you know approximately how far off that is at this point?

**Comment by gioxx** [comment]
We hope within the first half of June, if we can migrate to the new Manifest V3 and introduce the saving of the new Chrome Tab Groups (to avoid that you users keep losing them at each new update).

**Comment by Beskytt** [comment]
Hi - I'm having an issue with "Not Now", but weirdly enough it only has happened on a couple of pages so far. I navigate to another tab and then come back and still nothing happens when I click "Not Now".

I'll keep an eye out if it happens other places, but so far I've only seen it here: https://www.theurbanmuse.com/contact

And, ironically, it happens on this page that I'm using right now: https://github.com/gioxx/MarvellousSuspender/issues/86

TMS: 7.1.6.2
Chrome: 91.0.4472.77
OS & Version: Windows 10 Home 21H1

Pausing suspension works on other tabs, but this is a weird thing that hopefully will be fixed with the 7.1.6.3 update?

Thanks!

**Comment by PepeParada** [comment]
> > ¿Hay alguna forma de instalar la actualización ahora desde Github? No soy particularmente 'técnico' (como parece ser la mayoría de los usuarios aquí) pero puedo seguir las instrucciones si el procedimiento es simple.
> 
> ¡Por supuesto!
> Si desea utilizar la versión de GitHub, debe desinstalar la versión "pública" (Chrome Store) antes. **Por favor, preste atención** : anule la suspensión de todas las pestañas o exporte una copia de seguridad antes de la desinstalación, de lo contrario perderá todas las pestañas.
> 
> Ahora puede descargar el TMS actual (actualizado en GitHub) a través de https://github.com/gioxx/MarvellousSuspender/archive/refs/heads/master.zip . Descomprima el archivo y cargue una extensión descomprimida en Chrome:
> 
> * Abra una nueva pestaña en Chrome y vaya a chrome: // extensions
> * Habilitar el modo de desarrollador (en la esquina superior derecha)
> * Cargar extensión desempaquetada (botón en la esquina superior izquierda)
> * Cargue la carpeta "src" que ha descomprimido previamente (solo la carpeta "src", no toda la carpeta TMS GitHub)
> 
> Ahora tiene TMS 7.1.6.2 instalado desde GitHub, la versión modificada (en realidad un trabajo en progreso), puede suspender todas las pestañas nuevamente o (si ha perdido algunas pestañas), puede importar la sesión guardada
> 

Hello. Reading this comment was that I was able to find the solution to the problem I had. 
Thanks. Now it works perfectly.

Right now I have another question, which has nothing to do with this extension, but that maybe someone can solve, since I don't think I'm the only one affected: Chrome automatically removes the extension, after some time. 
I could not determine if it is just by restarting the browser, or if it is after restarting the computer.
I repeat, it is surely something that affects all manually added extensions, and not this one in particular. I do not know.
If someone knows and can give me an answer, I thank them. And again thank you and congratulate you for the excellent work and the attention provided.

**Comment by baz357** [comment]
> Right now I have another question, which has nothing to do with this extension, but that maybe someone can solve, since I don't think I'm the only one affected: Chrome automatically removes the extension, after some time.

It removes only the GitHub version you installed? Weird! Never happened here with _any_ extension (other than when Chrome decided that something was malware, but then it was only permanently disabled IIRC, not removed).

**Comment by PepeParada** [comment]
> > Ahora mismo tengo otra pregunta, que no tiene nada que ver con esta extensión, pero que tal vez alguien pueda resolver, ya que no creo que yo sea el único afectado: Chrome elimina automáticamente la extensión, después de un tiempo.
> 
> ¿Elimina solo la versión de GitHub que instaló? ¡Extraño! Nunca sucedió aquí con _ninguna_ extensión (excepto cuando Chrome decidió que algo era malware, pero luego solo se desactivó permanentemente IIRC, no se eliminó).

Very rare. It happened again. I rebooted the computer, and when I came back, at some point MSuspender stopped working.
TMS keeps showing up under "extensions", but the extension cannot be accessed because "it has been moved or deleted".
So, I go to the "\ Chrome \ User Data \ Default \ Extensions" folder and the unzipped folder is no longer there. 

screenshots:
![tms1](https://user-images.githubusercontent.com/52550955/122455372-b25bf080-cf82-11eb-906d-c51e105bb412.png)
![tms2](https://user-images.githubusercontent.com/52550955/122455482-d3244600-cf82-11eb-854b-09b15255feba.png)

(I translate with google. sorry for the brutality.).

**Comment by Beskytt** [comment]
> Hi - I'm having an issue with "Not Now", but weirdly enough it only has happened on a couple of pages so far. I navigate to another tab and then come back and still nothing happens when I click "Not Now".
> 
> I'll keep an eye out if it happens other places, but so far I've only seen it here: https://www.theurbanmuse.com/contact
> 
> And, ironically, it happens on this page that I'm using right now: #86
> 
> TMS: 7.1.6.2
> Chrome: 91.0.4472.77
> OS & Version: Windows 10 Home 21H1
> 
> Pausing suspension works on other tabs, but this is a weird thing that hopefully will be fixed with the 7.1.6.3 update?
> 
> Thanks!

Just wanted to bump this by saying that, currently, "Not Now" is doing the same behavior described above (that is, nothing happens when I click it) on almost all, if not all, pages.

TMS: v7.1.6.2 (same as before)
Chrome: 91.0.4472.114
OS & Version: Windows 10 Home 21H1 (same as before)

@gioxx Any thoughts?

**Comment by gioxx** [comment]
Ciao @Beskytt, we know the bug and we have also solved that on the GitHub version of TMS (https://github.com/gioxx/MarvellousSuspender/issues/86#issuecomment-851271379), public available for all in the next Chrome Store version.

**Comment by Beskytt** [comment]
@gioxx Gotcha - sorry, missed that bit! Any rough idea when the next Chrome Store release is coming? That is, is it worth it for me to get into messing around with the GitHub version now (which I haven't done before) or should I just hold off because the release will be out in the next like week or so?

**Comment by gioxx** [comment]
Good question. We are ready with some fixes and we would like to release to all using Chrome Store, but we are still working on the new Chrome "Tab Group" save function and Manifest V3 (take a look here: https://github.com/gioxx/MarvellousSuspender/issues?q=is%3Aopen+is%3Aissue+milestone%3A7.1.6.3).
It would be really convenient and nice to finish this change before releasing 7.1.3 to avoid further problems in the future, but at this point we're not sure we can release the new version on the Chrome Store before July.

**Comment by Beskytt** [comment]
@gioxx Understood. Not telling you how to do your job at all, but I wonder if it makes sense to release a small patch which addresses use-inhibiting bugs like this one (which might encourage user abandonment) even if you don't get to include the new functionality at the same time. I'm sure you have better data on this, but for my part I've been using "Not Now" all the time for sites I want to keep open but where I don't want to have to remember to remove them from a permanent whitelist (such as when I'm comparison shopping online).

Again, just my two cents. Love the extension!

**Comment by baz357** [comment]
> @gioxx Understood. Not telling you how to do your job at all, but I wonder if it makes sense to release a small patch which addresses use-inhibiting bugs like this one (which might encourage user abandonment) even if you don't get to include the new functionality at the same time.

Yes, PLEASE! :) @gioxx

**Comment by gioxx** [comment]
@Beskytt I absolutely agree with you, but we have received a lot of comments and negative reviews about the update mechanism that we are changing in the version that will be released soon (https://github.com/gioxx/MarvellousSuspender/issues/84).
It's really so hard to please everyone 😅

**Comment by Beskytt** [comment]
@gioxx Sure, I've worked with software communities a bunch, so I totally hear you on that.

... but maybe you could just please ME? 🤣 🤣 🤣 jk jk

I'm sure the next release will be great, I'm just obsessive about keeping a tidy whitelist on things like this, is all. For what it's worth, I've never noticed that I've been "nagged" about updates from TMS, it's felt pretty seamless. People love to complain and stamp their feet and complain about stuff they get for free all the time, but besides this one bug I've run into it's been A+.

**Comment by mkz7** [comment]
**@gioxx** Noting that it's a month since the last post and "Not Now" is still not working in Chrome (Brave), is there an expected 'arrival' date for that functionality to be back (without having to attempt installing the updated version in Github)?
Thanks.

**Comment by gioxx** [comment]
@mkz7 we are trying to close the gap to cover Chrome "tab group" functionality. If we solve this, we can push the new version of TMS on Chrome Store. I don't have an arrival date for sure, but we certainly try to upload a new version during the summer (with or without tab group feature).

**Comment by baz357** [comment]
> @mkz7 we are trying to close the gap to cover Chrome "tab group" functionality. If we solve this, we can push the new version of TMS on Chrome Store. I don't have an arrival date for sure, but we certainly try to upload a new version during the summer (with or without tab group feature).

I can only speak for myself, but at this point I don't care about new functions at all, I just want the extension to work as flawlessly and reliably as before. Thank you for continuing to work on it.

**Comment by mkz7** [comment]
@gioxx Thanks. Since the "Not now" fix was apparently taken care of quite a while ago, is it a problem to make an update available through the Chrome Store now and a later update with the 'tab group' feature available when that's ready?

**Comment by gioxx** [comment]
@mkz7 I personally have no problem releasing a new update on the Chrome Store, even if only to fix a bug. Unfortunately though, as already reported in this comment (https://github.com/gioxx/MarvellousSuspender/issues/86#issuecomment-864103013) I find myself having to suffer complaints due to the update that goes to disturb the activity of users. For this reason I would like to release one that is as complete and stable as possible and that already corrects several aspects of TMS.

**Comment by ossilator** [comment]
@gioxx have you tried making the upgrade less of a PITA? i already discussed that back with dean, and he liked my idea(s) (find them in the TGS tracker). i thought he implemented it already, but apparently not (fully)?

**Comment by 71walceli** [comment]
It's weird, amd even more so b'cuz of the workaround of switching back and forth the tabs makes it work. Also, the same behavior seems to happen when clicking "Pause/unpause suspension of this tab" from TMS submenu in right-click context menu...

**Comment by mkz7** [comment]
@gioxx What I am noticing is that the '_switch-to-another-tab-then-back-to-the-tab-you-want-temporarily-suspended_' trick (i.e., the workaround to activate 'Not now') does not seem to actually be doing that.

The tab's icon grays out when that sequence is performed, making it appear that the tab won't suspend. However, checking the status of the tab (by clicking on the TMS icon), I find that "Tab will suspend automatically. Not now" still displays.

So the actual status of the tab via the workaround is unclear. If it is actually already in a 'Not now' state, the option to select that should not display.

**Comment by witcheries** [comment]
I'm also experiencing this issue, although the status of the tab's suspension seems to be accurate, ie the icon is greyed out and the status reads as 'Tab suspension paused. Unpause' when clicking on the icon. 

@gioxx I just wanted to check on the status of the update to the Chrome Web Store version. The last timing estimate was 'during the summer' and now it's almost September. Not rushing y'all at all, just curious about an updated time estimate. I can figure out how to install the GitHub version if the CWS update is still a ways off.

**Comment by mkz7** [comment]
@gioxx @witcheries 

Just to be clear, the 'switch away and then back' process that it was claimed would achieve the 'Not now' state was said to be induced by the following sequence:

Click on 'Not now' -- nothing happens.
Click on another active tab in the same window.
Return to the first tab. Click again on 'Not now'. 
This time the 'Not now' condition displays as active.

Is that correct? (That's what I'm finding is **not** happening.)

**Comment by gioxx** [comment]
ciao @mkz7 and @witcheries, the problem is manifest v3, unfortunately we've suffered a slowdown in the migration and @napolux is overwhelmed with work, which is why we haven't been able to publish an updated version of TMS for the Chrome Store that solves the reported problems and introduces those new features we've been talking about. Although I understand that an "intermediate" release is certainly welcome, I'm convinced that the novelty regarding the update system (that doesn't disturb you users in case of updates) is fundamental to land on the Chrome Store with the new version.

Feel free to install the GitHub version, while waiting the good news about the next official and stable version released on the Chrome Store.

Thanks (really) for all the support and patience.
Giovanni.

**Comment by digitalambition** [comment]
I'm sorry to raise this again, but it's mid-October and "Not Now" - a feature I use(d to use) on a daily basis - still isn't working, and finding this thread is the only reason I didn't uninstall the extension today. I'm happy to continue being patient for an approximately-known timeframe, if you have one, for the Chrome Store? Otherwise I'll probably join others (above) in switching to the Github version.

Thank you for your work in creating this extension, for making it open source, and in somehow remaining positive and friendly in response to feedback -- even when some of the commenters haven't really deserved it!

**Comment by d4h0** [comment]
@digitalambition: What I'm doing at the moment is to just use the 'Never suspend this URL' feature (which works). From time-to-time I go to the settings page and delete everything from the 'Never suspend tabs with URLs from the following list' text area. This works well enough for me at the moment, but I still would love to get the 'Not now' feature back... :)

**Comment by mkz7** [comment]
Ciao @gioxx . . . Was TMS 7.1.6.3 ever uploaded to the Chrome Store?

**Comment by mkz7** [comment]
@gioxx In this post from last August -- **https://github.com/gioxx/MarvellousSuspender/issues/86#issuecomment-900687363** -- I mentioned that the issue wasn't limited to 'Not now' not working when clicked, but that the temporary solution of clicking another tab and then clicking back to the original tab also does not work. I can document that what I reported then continues to occur, with an example:

– I select 'Not now' for a tab. It has no effect. The TMS icon remains 'brown' (i.e., the tab will 'suspend' according to currently-selected 'suspend' setting).
– I click on another tab in the same window. That tab's window displays.
– I then click again on the first tab. The TMS icon immediately changes from 'brown' to 'gray'. 

The gray normally indicates a 'whitelisting' state (meaning the tab should not 'suspend'). Given the sequence mentioned above, it should now be in the 'Not now' state, not the 'Never' state. Note, however, that the icon, though gray, when clicked still offers 'Not now' -- which normally would mean the tab is **not whitelisted** and will suspend according to the default (or otherwise currently-selected) 'suspend' setting. 

– Within several minutes, the gray tab reverts to 'brown'. This does take several minutes to occur -- i.e., the icon temporarily displays as gray per the previous step, but invariably it does revert to brown after a delay. This reflects the actual state of the tab. No ability to induce 'Not now' through tab switching actually works.

**Comment by witcheries** [comment]
> @gioxx 
> 
> Just to be clear, the 'switch away and then back' process that it was claimed would achieve the 'Not now' state was said to be induced by the following sequence:
> 
> Click on 'Not now' -- nothing happens. Click on another active tab in the same window. Return to the first tab. Click again on 'Not now'. This time the 'Not now' condition displays as active.
> 
> Is that correct? (That's what I'm finding is **not** happening.)

That's correct, although the second tab didn't need to be live. It could also be suspended, and the workaround would still work. However, I'm now noticing that the 'Not now' is working as expected, so I wonder if a fix was released?

**Comment by mkz7** [comment]
@witcheries 

_> I'm now noticing that the 'Not now' is working as expected, so I wonder if a fix was released?_

I'm not seeing any difference, myself. I click on "Not now' and nothing happens.

And as noted in the post you replied to, even the workaround was no longer working (which continues to be the case for me).

**Comment by gioxx** [comment]
> Ciao @gioxx . . . Was TMS 7.1.6.3 ever uploaded to the Chrome Store?

Ciao @mkz7, sorry for the late reply. No, TMS 7.1.6.3 has not been uploaded to the Chrome Store because I want to avoid the negative comments from those who would go and lose the tab groups, as mentioned several times.

I'm struggling with development because what's being imposed on us by Chrome's Manifest V3 is preventing us from working as we'd like, and we're not sure we can move TMS forward. That said, we will try to focus on saving Tab Groups so we can integrate it with Manifest V2 (I seem to remember it wasn't possible but we'll look at it again) and thus release a good updated version that allows everyone to work properly.

I don't have any release forecasts on this but we are doing what we can in the little spare time to get the task done. If there were more developers out there that could help us, that would be great and allow us to speed up.

Up to now the best solution maybe remains to use the version released here on GitHub (although I realize that you lose the convenience of letting Chrome update it by taking it directly from its Store).

**Comment by mkz7** [comment]
_> I want to avoid the negative comments from those who would go and lose the tab groups_

Thanks, @gioxx. I didn't follow the 'tab groups' issue but couldn't TMS users just be advised to use another extension that saves tab groups until you're able to implement that feature in an updated TMS? (e.g., they could use Tabli for that, which saves tab groups as bookmarks)

If I were to opt to use the Github version, is it necessary to 'uninstall' the current Chrome version or can it just be 'disabled' on the extensions page (red button on the far right)? If not, by 'uninstall' do you just mean use the 'remove' button?

Also, re the last instruction in your 'how to' post -- "**Load the "src" folder**" -- how is that done? I assume you don't do it by using the same "Load unpacked" button used for the prior instruction (or do you?) . . . https://github.com/gioxx/MarvellousSuspender/issues/86#issuecomment-851895055 

Thanks again.

**Comment by mkz7** [comment]
Hi @gioxx, 

I mentioned Tabli (another tab management extension) in my last post -- as a way that TMS users could save tab groups until you can incorporate that feature in TMS.

I've since recalled that **Tabli's developer had included support for The Great Suspender**. 

When the 'issues' with TGS happened, he indicated that he would do the same for any fork by a "trustworthy developer". So you could contact him to do that with TMS if you would like that to happen.

**Tabli provides 'bookmarking' of tab groups**, so the TMS users who want that feature could get it in an interim way from Tabli. You could then (if you want to) get a current update of TMS into the Chrome Store for your users, one that includes the 'Not now' fix, and leave the 'tab groups' issue for later.

Just a thought.

TABLI . . . https://chrome.google.com/webstore/detail/tabli/igeehkedfibbnhbfponhjjplpkeomghi 

Antony Courtney <antony@antonycourtney.com>
Tabli Dev Team <tabli-feedback@antonycourtney.com>

**Comment by CollinChaffin** [comment]
No offense, but it's coming up on a full year since this rather simple code fix was implemented in the github source via/unpack version, yet a year later a MAJOR bug like this still isn't pushed to the Chrome store? I can guarantee that the "complaints" you were facing by releasing this fix, then the other updates, has been FAR exceeded now by those using the webstore extension where one of the few features this extension offers - is 100% broken.  I'm curious what metric has you so convinced that releasing two releases, this hotfix, then whatever you've been working on for session mgmgt, would cause the sky to fall to that extent for TMS users?

BTW, I've been using TGS since day one, and NEVER use the session mgmt because for that, I use Session Buddy, which provides session mgmt to a level that TGS/TMS will never possibly achieve - and that's ALL it does. Anyone can install that, save their sessions import/export etc. and never need to worry about "losing" TMS sessions - it's what I do. ALL I want TGS/TMS to do, is tab suspension - and to do it well - that's it.

**Comment by githubjsorg** [comment]
This is still an issue for me as well.

**Comment by Technetium1** [comment]
@mkz7 Uninstallation of the web store version is not required, but disabling it is. If you install the GitHub release any currently suspended tabs will disappear, so be mindful of that. Unsuspend and load them all beforehand, then save the session before continuing with the installation.

Tablii already has an open issue for that here: https://github.com/antonycourtney/tabli/issues/159 and a pull request including TMS ID here: https://github.com/antonycourtney/tabli/pull/165

**Comment by mkz7** [comment]
> _@mkz7 Uninstallation of the web store version is not required, but disabling it is. If you install the GitHub release any currently suspended tabs will disappear, so be mindful of that. Unsuspend and load them all beforehand, then save the session before continuing with the installation._
 
> _Tablii already has an open issue for that here: [antonycourtney/tabli#159](https://github.com/antonycourtney/tabli/issues/159) and a pull request including TMS ID here: [antonycourtney/tabli#165](https://github.com/antonycourtney/tabli/pull/165)__

@Technetium1 Thanks, Courtney's posts date back quite a while. I doubt he is even involved at all with #159 and #165 at this point. 

Given the length of time the 'Not now' issue has remained outstanding, I've concluded that @gioxx is just not going to get to fixing it in the Chrome store version.

Appreciate the information (especially the warning) about the GitHub installation.

**Comment by mkz7** [comment]
> BTW, I've been using TGS since day one, and NEVER use the session mgmt because for that, I use Session Buddy, which provides session mgmt to a level that TGS/TMS will never possibly achieve - and that's ALL it does. Anyone can install that, save their sessions import/export etc. and never need to worry about "losing" TMS sessions.

@CollinChaffin Saving sessions in TMS seems straightforward enough. Can you say a bit about what makes Session Buddy superior for that purpose? 

In the past, every time I looked at the user posts regarding Session Buddy issues, there were a great many complaints about lack of response from the developer. Has that changed?

Thanks.

**Comment by I-am-the-Users** [comment]
Same :(
 * Extension version: 7.1.6.2
 * Browser name & version: Brave Version 1.42.97 | Chromium: 104.0.5112.102 (Official Build) (64-bit)
 * Operating system & version: Windows 11 Pro Version 21H2
---
Other similar extensions like
* The Great Suspender | ID: jaekigmcljkkalnicnjoafgfjoefkpeg
* The Great Suspender Original | ID: ahmkjjgdligadogjedmnogbpbcpofeeo

Works fine.

**Comment by mGER86** [comment]
*PUSH*

same Problem

- Extension version: **7.1.6.2**
- Browser name & version: **Google Chrome 113.0.5672.93 (Official Build) (64-bit)**
- Operating system & version: **Windows 11 Home Version 21H2 (Build 22000.1936)**

![suspend](https://github.com/gioxx/MarvellousSuspender/assets/133798420/2596a2ef-124e-424f-9ef2-b245fcecc75e)

**Comment by githubjsorg** [comment]
This is still an issue for me but it is inconsistent. Sometimes it works, other times it does not.

**Comment by sebas-benme** [comment]
Still having this on `v7.1.6.2`
Chrome: `120.0.6099.71 (Official Build) (64-bit)`
OS: `Windows 11 Version 23H2 (0S Build 22631.2861)`

**Comment by DynamosRose** [comment]
It's 2024 and the "not now" button has never worked for me, with or without workarounds

**Comment by xodotLBO** [comment]
> It's 2024 and the "not now" button has never worked for me, with or without workarounds

Used to work for me but now it stopped, unfortunate.

**Comment by welovfree** [comment]
I am stil facing this issue in version `v7.1.6.2`.
